app/models/application_record.rb

```diff
@@ -3,4 +3,21 @@
 # Abstract record class to be applied to all models
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+
+  def self.arel_find(collection, batch_size = 1000)
+    base_arel = arel_table[column_names.first].eq('')
+    collection.in_groups_of(batch_size, false).map do |batch|
+      where(batch.inject(base_arel) do |rel, record|
+        rel.or(yield(record))
+      end)
+    end.inject([], &:+)
+  end
+
+  def ==(other)
+    return super if self.class.column_names.include? 'id'
+
+    self.class.column_names.map do |col|
+      send(col) == other.send(col)
+    end.all?
+  end
```

Comment:
This method short circuits for normal tables with an ID field. For those, equality is based on their ids matching. For join tables, such as RuleReferencesRule, there is no ID, so the records are never equal. This method instead compares their column values, which for a join table is simply two IDs (i.e. rule_id and rule_reference_id).

Reply:
Are there any performance penalties when it tries to compare every column value? 
Maybe this could go into a concern and only include it for join tables?

Reply:
For most models (those with an 'id' column), this changes nothing. Notice that it short circuits to `super` on line 17.